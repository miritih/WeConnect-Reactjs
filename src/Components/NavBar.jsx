import React from 'react';
import { Link } from 'react-router-dom';
import Authservice from './Auth/AuthService';
import { toast } from 'react-toastify';
import { Image } from 'cloudinary-react';
import { cloudName } from '../utils/Config';

class NavBar extends React.Component {
	constructor() {
		super();
		this.state = {
			isloggedin: false,
			user: []
		};
		this.Auth = new Authservice();

	}
	handleLogout(e) {
		e.preventDefault();
		this.Auth.logout();
		this.setState({
			isloggedin: this.Auth.isLoggedIn()
		});
		this.props.history.replace('/');
		toast.success(({ closeToast }) => <div>
			<h3>Success</h3>
			<p>You have been successfully Logged out</p>
		</div>);
	}
	componentDidMount() {
		this.setState({
			isloggedin: this.Auth.isLoggedIn(),
			user: this.Auth.getUser()['user']
		});
	}
	render() {
		return (
			<div>
				<nav className={
					('/' === this.props.location.pathname ?
						'navbar navbar-expand-lg navhome navbar-light bg-transparent fixed-top' :
						'navbar navbar-expand-lg navbar-light fixed-top navbarcolor'
					)}>
					<div className="container">
						<Link className="navbar-brand" to="/">
							<span className={(
								'/' === this.props.location.pathname ?
									'white' : ''
							)}>WeConnect</span>
						</Link>
						<button className="navbar-toggler" type="button" data-toggle="collapse"
							data-target="#topnav" aria-controls="topnav" aria-expanded="false"
							aria-label="Toggle navigation">
							<span className="navbar-toggler-icon"></span>
						</button>
						<div className="collapse navbar-collapse" id="topnav">
							<ul className={'navbar-nav ml-auto ' + ('/' === this.props.location.pathname ? 'white' : '')}>
								<li className="nav-item">
									<Link className="nav-link" to="/">Home</Link>
								</li>
								{this.state.isloggedin ?
									<span>
										<li className="nav-item">
											<Link className="nav-link" to="/register_business">Register business</Link>
										</li>
									</span>
									:
									<li className="nav-item">
										<Link className="nav-link" to="/register">Signup</Link>
									</li>
								}
								{this.state.isloggedin ?

									<li className="nav-item dropdown">
										<a className="nav-link dropdown-toggle text-capitalize user-profile" href="avascript:;" id="navbarDropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
											<Image cloudName={cloudName} publicId={this.state.user['image']} />
											{' ' + this.state.user['username']}
										</a>
										<div className="dropdown-menu" aria-labelledby="navbarDropdownMenuLink">
											<Link className="dropdown-item" to="/profile/update">Manage Profile</Link>
											<Link className="dropdown-item" to="/logout" onClick={this.handleLogout.bind(this)}>Logout</Link>
										</div>
									</li>
									:
									<li className="nav-item">
										<Link className="nav-link" to="/login">Login</Link>
									</li>
								}
							</ul>
						</div>
					</div>
				</nav>
			</div>
		);
	}
}

export default NavBar;
