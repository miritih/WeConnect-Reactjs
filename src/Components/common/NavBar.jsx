import React from 'react';
import { Link } from 'react-router-dom';
import NavLinks from './NavLinks';
import Authservice from '../Auth/AuthService';
const Auth = new Authservice();

class NavBar extends React.Component {
	constructor(props) {
		super(props);
		this.handleLogout = this.handleLogout.bind(this);
	}
	handleLogout(e) {
		e.preventDefault();
		Auth.logout();
		this.props.actions.isLoggedIn(); //sets logged in state to false after logout
		this.props.history.replace('/'); // redirect to home page
	}
	render() {
		return (
			<div>
				<nav className={
					// apply transparent bg if on homepage
					('/' === this.props.location.pathname ?
						'navbar navbar-expand-lg navhome navbar-light bg-transparent fixed-top' :
						'navbar navbar-expand-lg navbar-light fixed-top navbarcolor'
					)}>
					<div className="container">
						<Link className="navbar-brand" to="/">
							<span className={( //if on homepage set brand color to white
								'/' === this.props.location.pathname ?
									'white' : ''
							)}>WeConnect</span>
						</Link>
						<button className="navbar-toggler" type="button" data-toggle="collapse"
							data-target="#topnav" aria-controls="topnav" aria-expanded="false"
							aria-label="Toggle navigation">
							<span className="navbar-toggler-icon"></span>
						</button>
						<div className="collapse navbar-collapse" id="topnav">
							{/* if on homepage set navlinks color to white */}
							<ul className={'navbar-nav ml-auto ' + ('/' === this.props.location.pathname ? 'white' : '')}>
								<NavLinks
									loggedIn={this.props.loggedIn}
									location={this.props.location}
									user={this.props.user}
									actions={this.props.actions}
									handleLogout={this.handleLogout}
								/>
							</ul>
						</div>
					</div>
				</nav>
			</div>
		);
	}
}
export default NavBar;