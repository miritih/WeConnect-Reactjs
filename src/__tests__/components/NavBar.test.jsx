import React from 'react';
import NavBar from '../../Components/NavBar';
import Authservice from '../../Components/Auth/AuthService';
import { mount } from 'enzyme';
// import react router from react-router-dom
import { MemoryRouter as Router } from 'react-router-dom';
// import renderer for snapshot testing
import renderer from 'react-test-renderer';

describe('NavBar', () => {

  let wrapper;
  const pathname = {
    pathname: "/login"
  };

  it('renders without crashing', () => {
    const rendered = renderer.create(
      <Router><NavBar location={pathname} /></Router>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('should call handlesubmit when form is submitted', () => {
    const handleLogout = jest.fn();
    wrapper = mount(<Router><NavBar location={pathname}/></Router>);
    const li = wrapper.find('ul').childAt(2);
    li.simulate('click');
    expect(handleLogout).toBeCalled;
  });
});
