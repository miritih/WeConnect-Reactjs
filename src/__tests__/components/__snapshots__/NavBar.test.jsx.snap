// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders without crashing 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light fixed-top navbarcolor"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    <span
      className=""
    >
      WeConnect
    </span>
  </a>
  <button
    aria-controls="topnav"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#topnav"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="topnav"
  >
    <ul
      className="navbar-nav ml-auto "
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/register"
          onClick={[Function]}
        >
          Signup
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
      
    </ul>
  </div>
</nav>
`;
